### PR TITLE
PostgreSQL operator returns query result 

### LINKED ISSUE #15010: Allow PostgreSQL's operator to return the query result
**Description**

Right now this operator is not able to return any query result

**Use case / motivation**

It should be allowed to do so in order to use them to retrieve useful data and use it in the following nodes

**Are you willing to submit a PR?**

Yup ofc!
See https://github.com/apache/airflow/pull/15012 as a POC to start a discussion about the feasibility of the proposal

### PR DESCRIPTION
**Pull request by ghost**
closes: #15010

Right now this operator is not able to return any query result.
It should be allowed to do so in order to use them to retrieve useful data and use it in the following nodes

### CODE REVIEW

File: airflow/providers/postgres/operators/postgres.py [hunk 1]
```diff
@@ -68,6 +68,7 @@ def __init__(
     def execute(self, context):
         self.log.info('Executing: %s', self.sql)
         self.hook = PostgresHook(postgres_conn_id=self.postgres_conn_id, schema=self.database)
-        self.hook.run(self.sql, self.autocommit, parameters=self.parameters)
+        result = self.hook.run(self.sql, self.autocommit, parameters=self.parameters)
         for output in self.hook.conn.notices:
             self.log.info(output)
+        return result
```
Comment (ashb) [inline airflow/providers/postgres/operators/postgres.py]:
```suggestion
        if self.do_xcom_push:
            return result
```

**Review by ashb — CHANGES_REQUESTED** [review]
👍🏻 

Please add some unit tests too.

**Comment by marcosmarxm** [comment]
The `DBApiHook` **run** hasn't any return. Also reading the documentation:
 ```
 Runs a command or a list of commands. Pass a list of sql
        statements to the sql parameter to get them to execute
        sequentially
```
you can send a list of sql statements to run, how do you proceed with this?
Maybe some logic using the function `get_records` could be better...

This look a very nice think to use with the TaskFlow API. Others users must know about this. Please, if possible add example and improve documentation to demonstrate to others.

**Comment by nttdriva** [comment]
Hi @marcosmarxm 

> The DBApiHook run doesn't return anything
Method overriding in Python is perfectly fine with different return types when using inheritance. I guess that the fact that Python does not support overloading helps preventing issues.

> you can send a list of sql statements to run, how do you proceed with this?
I'll make some tests and I'll get back to you.

Thank you for the feedback

**Comment by Davide95** [comment]
Update: I was not able to reproduce the steps provided here (https://github.com/apache/airflow/blob/master/LOCAL_VIRTUALENV.rst) after some hours of attempts.
I'm using the custom version of this operator and it works fine, but I do not have much spare time these days to fix the issues related to the dev env and to make the required tests.
I'll keep this PR opened in the case someone wants to contribute

**Pull request closed without merging**